src/System.Private.ServiceModel/tests/Scenarios/SelfHostWcfService/TestResources/EndpointResource.cs

```diff
@@ -32,10 +32,22 @@ public object Put()
                     {
                         host = new ServiceHost(typeof(ServiceType));
                         // URI assumes localhost but that instead should be passed in
+                        // For Kerberos, localhost cannot be used as localhost will fall back to NTLM
+                        // this is a workaround until the URI is passed in
+                        Binding binding = GetBinding();
+                        string baseAddress = "localhost";
+                        if (binding is BasicHttpBinding)
```

Comment:
I think it should be fixed now. The issue for passing in the URL should not also be responsible for refactoring this code.  It means we need a virtual method to compose the URI -- and that is what the URI-passing issue expects to fix.